codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -557,7 +709,8 @@ int incrementStreamWindow(int delta) throws Http2Exception {
                 throw streamError(stream.id(), FLOW_CONTROL_ERROR,
                         "Window size overflow for stream: %d", stream.id());
             }
-            int previouslyStreamable = streamableBytes();
+            final int previouslyStreamable = streamableBytes();
+            final int previousWritableWindow = writableWindow();
```

Comment:
Maybe rename `previouslyStreamableBytes` to `previouslyStreamableBytes`.  Then make the flag`previouslyWritable = writableWindow() > 0`.

Then below, the code can be simplified to:

```java
if (listener != null) {
    boolean currentlyWritable = writableWindow() > 0;
    writabilityChanged = previouslyWritable ^ currentlyWritable;
}
```